From e56ec39e66f70c1da4425dde15247c7e9daffd57 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Tue, 12 Nov 2024 10:41:14 +0100
Subject: [PATCH 41/55] busybox: do not use pool servers if DHCP provided NTP

Disables default NTP pool servers when DHCP provides NTP server addresses, avoiding redundant time sources and reducing network traffic.

Signed-off-by: John Crispin <john@phrozen.org>
---
 package/utils/busybox/files/sysntpd | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/package/utils/busybox/files/sysntpd b/package/utils/busybox/files/sysntpd
index 074f14b8f8..43b8b6eec5 100755
--- a/package/utils/busybox/files/sysntpd
+++ b/package/utils/busybox/files/sysntpd
@@ -18,6 +18,8 @@ get_dhcp_ntp_servers() {
 
 	ntpservers=$(ubus call network.interface dump | jsonfilter -e "@.interface[$filter]['data']['ntpserver']")
 
+	[ -n "$ntpservers" ] && server=""
+
 	for ntpserver in $ntpservers; do
 		local duplicate=0
 		local entry
-- 
2.34.1

